src/platform/src/test/java/org/geoserver/util/XmlTestUtil.java

```diff
@@ -0,0 +1,200 @@
+     * @param xPath
+     * @param matcher
+     * @return
+     */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public Matcher<Document> hasOneNode(final String xPath, final Matcher<? super Node> matcher) {
+        return hasNodes(xPath, (Matcher)contains(matcher));
+    }
+    /**
+     * Match a document where one node matches the XPath expression.
+     * @param xPath
+     * @param matcher
+     * @return
+     */
+    public Matcher<Document> hasOneNode(final String xPath) {
+        return hasOneNode(xPath, any(Node.class));
+    }    
+    /**
+     * Match a document at least one of the nodes matched by the given XPath expression matches the given matcher.
+     * @param xPath
+     * @param matcher
+     * @return
+     */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public Matcher<Document> hasNode(final String xPath, final Matcher<Node> matcher) {
+        return hasNodes(xPath, (Matcher) hasItem(matcher));
+    }
+    
+    /**
+     * Match a document at least one node matches the given XPath.
+     * @param xPath
+     * @param matcher
+     * @return
+     */
+    public Matcher<Document> hasNode(final String xPath) {
+        return hasNode(xPath, any(Node.class));
+    }
+    
+    /**
+     * Match a document where the list of nodes selected by the given XPath expression also matches the given matcher.
+     * @param xPath
+     * @param matcher
+     * @return
+     */
+    public Matcher<Document> hasNodes(final String xPath, final Matcher<? extends Iterable<Node>> matcher) {
+        return new BaseMatcher<Document>() {
+            
+            @Override
+            public boolean matches(Object item) {
+                XpathEngine engine = XMLUnit.newXpathEngine();
+                engine.setNamespaceContext(namespaceContext);
+                try {
+                    List<Node> nodes = nodeCollection(engine.getMatchingNodes(xPath, (Document) item));
+                    return matcher.matches(nodes);
+                } catch (XpathException e) {
+                    return false;
+                }
+            }
+            
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("Document where the list of nodes matching ")
+                    .appendValue(xPath).appendText(" is ").appendDescriptionOf(matcher);
+            }
+            
+            @Override
+            public void describeMismatch(Object item, Description description) {
+                XpathEngine engine = XMLUnit.newXpathEngine();
+                engine.setNamespaceContext(namespaceContext);
+                try {
+                    List<Node> nodes = nodeCollection(engine.getMatchingNodes(xPath, (Document) item));
+                    
+                    matcher.describeMismatch(nodes, description);
+                    
+                    if(showXML != null) {
+                        printDom((Document) item, System.out);
```

Comment:
Yes, I'd initially made showXML a boolean then changed it to a stream for more flexibility.  I forgot to update this.